service/frontend/workflowHandler.go

```diff
@@ -918,6 +869,10 @@ func (wh *WorkflowHandler) RespondWorkflowTaskFailed(
 
 	defer log.CapturePanic(wh.GetLogger(), &retError)
 
+	if wh.isStopped() {
```

Comment:
It would be very nice if we could do this in one of the interceptor.

Reply:
Yes, I also noticed this. And `versionChecker` is another candidate. Will do it separately.